controllers/imageclusterinstall_controller_test.go

```diff
@@ -903,20 +895,101 @@ var _ = Describe("Reconcile", func() {
 		Expect(c.Create(ctx, clusterInstall)).To(Succeed())
 		Expect(c.Create(ctx, clusterDeployment)).To(Succeed())
 
+		key := types.NamespacedName{
+			Namespace: bmh.Namespace,
+			Name:      bmh.Name,
+		}
 		req := ctrl.Request{
 			NamespacedName: types.NamespacedName{
 				Namespace: clusterInstallNamespace,
 				Name:      clusterInstallName,
 			},
 		}
+
 		res, err := r.Reconcile(ctx, req)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(res).To(Equal(ctrl.Result{}))
 
+		Expect(c.Get(ctx, key, bmh)).To(Succeed())
+		Expect(bmh.Spec.Image.URL).To(Equal("https://images-namespace.cluster.example.com/images/test-namespace/test-cluster.iso"))
+		resourceVersion := bmh.ResourceVersion
+
+		By("Nothing was changed in the clusterInstall, so no bmh change should happen")
+		res, err = r.Reconcile(ctx, req)
+		Expect(err).NotTo(HaveOccurred())
+
+		res, err = r.Reconcile(ctx, req)
+		Expect(err).NotTo(HaveOccurred())
+		res, err = r.Reconcile(ctx, req)
+		Expect(err).NotTo(HaveOccurred())
+		res, err = r.Reconcile(ctx, req)
+		Expect(err).NotTo(HaveOccurred())
+		res, err = r.Reconcile(ctx, req)
+		Expect(err).NotTo(HaveOccurred())
+		res, err = r.Reconcile(ctx, req)
+		Expect(err).NotTo(HaveOccurred())
+		res, err = r.Reconcile(ctx, req)
+		Expect(err).NotTo(HaveOccurred())
+		res, err = r.Reconcile(ctx, req)
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
This is a bit excessive I think

Reply:
oops :) actually by mistake, it was part of my manual testing